nupic/research/frameworks/dendrites/utils.py

```diff
@@ -59,4 +62,23 @@ def plot_dendritic_activations(
     ]
     y_labels = ["dendrite {}".format(j) for j in range(num_dendrites)]
 
-    return wandb.plots.HeatMap(x_labels, y_labels, activations, show_text=False)
+    # Use matplotlib to plot the activation heatmap
+    fig, ax = plt.subplots()
+    ax.imshow(activations, cmap="coolwarm_r")
+
+    ax.set_xticks(np.arange(num_contexts))
+    ax.set_yticks(np.arange(num_dendrites))
+
+    ax.set_xticklabels(x_labels)
+    ax.set_yticklabels(y_labels)
+
+    plt.setp(ax.get_xticklabels(), rotation=45, ha="right", rotation_mode="anchor")
+    plt.tight_layout()
+
+    # Annotate just the top absolute activation for each context
+    top_activation_dendrite_per_context = np.argmax(np.abs(activations), axis=0)
+    for j, i in enumerate(top_activation_dendrite_per_context):
+        val = np.round(activations[i, j], 2)
+        ax.text(j, i, val, ha="center", va="center", color="w")
+
+    return plt
```

Comment:
There are several other ways to return what's needed to log your plot to wandb. In fact, wandb accepts many plot-like things. In this case, you could return `fig` or `ax` (I'd recommend the `fig`).

To illustrate the flexibility of what wandb accepts, here are some examples:
```
# %%

import wandb
import matplotlib.pyplot as plt

# Init wandb
wandb.init(name="line_plot", project="test_wandb")

# Give wandb the pyplot module.
plt.plot([1, 2, 3], [1, 2, 3])
plt.title("my_plot")
wandb.log({"plt": plt})

# Give wandb the line object returned.
lines = plt.plot([1, 2, 3], [1, 2, 3])
line = lines[0]  # there's only one in the list
wandb.log({"line": line})  # log line object to wandb

# Give wandb the current figure.
plt.plot([1, 2, 3], [1, 2, 3])
plt.title("my_plot")
fig = plt.gcf()  # get current figure
wandb.log({"fig": fig})


# Given wandb the current axis.
plt.plot([1, 2, 3], [1, 2, 3])
plt.title("my_plot")
ax = plt.gca()  # get current axis
wandb.log({"ax": ax})
```

Note, I was somewhat confused why one could use either `fig` or `ax`. How can these be treated by `wandb` as the same thing? This [link](https://realpython.com/python-matplotlib-guide/#:~:text=A%20Figure%20object%20is%20the,%E2%80%9D%20as%20we%20might%20expect) helped. It seems `ax` is actually a nested object within `fig`.  In the end, wandb just wants the figure, but it can get it by multiple means. It can access `ax.figure` or it can ask the module for the current figure as in `plt.gcf()`. Here's a distilled version of their logic:
```
def get_matplotlib_figure(obj):
    if obj == matplotlib.pyplot:
        obj = obj.gcf()
    elif not isinstance(obj, Figure):
        if hasattr(obj, "figure"):
            obj = obj.figure
       else:
            raise ValueError
    return obj
```

Reply:
Thanks for looking into this, I decided to go with `plt.gcf()` and return that value.

Reply:
That works. Although I believe you could have skipped the call to `plt.gcf()` and returned `fig` as it's defined earlier in the function.